python/ffsim/tenpy/hamiltonians/molecular_hamiltonian.py

```diff
@@ -0,0 +1,45 @@
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""TeNPy molecular Hamiltonian."""
+
+from __future__ import annotations
+
+from tenpy.models.molecular import MolecularModel
+
+from ffsim.hamiltonians.molecular_hamiltonian import MolecularHamiltonian
+
+# ignore lowercase variable checks to maintain TeNPy naming conventions
+# ruff: noqa: N806
+
+
+class MolecularHamiltonianMPOModel(MolecularModel):  # type: ignore
```

Comment:
We don't need this class anymore, let's just use MolecularModel from TeNPy directly.